scripts/plot_dispatch.py

```diff
@@ -0,0 +1,130 @@
+    plt.xticks(fontsize=14)
+    plt.yticks(fontsize=14)
+    # format x-axis representing the dates
+    plt.gca().xaxis.set_major_formatter(mdates.DateFormatter("%Y-%m-%d"))
+    plt.gca().xaxis.set_major_locator(mdates.WeekdayLocator())
+
+    # Shrink current axis's height by 10% on the bottom
+    box = ax.get_position()
+    ax.set_position(
+        [box.x0, box.y0 + box.height * 0.15, box.width, box.height * 0.85]
+    )
+    # Put a legend below current axis
+    ax.legend(
+        loc="upper center",
+        bbox_to_anchor=(0.5, -0.1),
+        fancybox=True,
+        ncol=4,
+        fontsize=14,
+    )
+
+    fig.tight_layout()
+    return fig
+
+
+if __name__ == "__main__":
+
+    paths = Dict()
+    paths.postprocessed = sys.argv[1]
+    paths.plotted = sys.argv[2]
+
+    # create the directory plotted where all plots are saved
+    if not os.path.exists(paths.plotted):
+        os.makedirs(paths.plotted)
+
+    timeseries_directory = os.path.join(paths.postprocessed, "oemoflex-timeseries")
+    timeseries_files = os.listdir(timeseries_directory)
+
+    # select timeframe
+    timeframe = [
+        ("2019-01-01 00:00:00", "2019-01-31 23:00:00"),
+        ("2019-07-01 00:00:00", "2019-07-31 23:00:00"),
+    ]
+
+    # possible carriers: "electricity.csv", "heat_decentral", "heat_central"
+    carriers = ["electricity.csv", "heat_decentral", "heat_central"]
+    # possible regions: "AT", "BE", "CH", "CZ", "DK", "DE", "FR", "IT", "LU", "NL", "PL"
+    regions = ["DE", "FR", "PL"]
+
+
+    selected_timeseries_files = [file for file in timeseries_files for carrier in carriers for region in regions
+                                 if carrier in file and region in file]
+
+    for bus_file in selected_timeseries_files:
+
+        bus_name = os.path.splitext(bus_file)[0]
+        bus_path = os.path.join(timeseries_directory, bus_file)
+
+        data = pd.read_csv(bus_path, header=[0, 1, 2], parse_dates=[0], index_col=[0])
+
+        # prepare dispatch data
+        # convert data to SI-unit
+        conv_number = 1000
+        data = data * conv_number
+        data = sum_demands(data, bus_name=bus_name, demand_name="demand")
+        df, df_demand = plots.prepare_dispatch_data(
+            data, bus_name=bus_name, demand_name="demand", labels_dict=plot_labels
+        )
+
+        # interactive plotly dispatch plot
+        fig_plotly = plots.plot_dispatch_plotly(
+            df=df,
+            df_demand=df_demand,
+            unit="W",
+            colors_odict=colors_odict
+        )
+
+        file_name = bus_name + "_dispatch_interactive" + ".html"
+        fig_plotly.write_html(
+            file=os.path.join(paths.plotted, file_name),
+            # include_plotlyjs=False,
+            # full_html=False
+        )
+
+
+        for start_date, end_date in timeframe:
+            fig = draw_plots(df=df, start_date=start_date, end_date=end_date, bus_name=bus_name, colors_odict=colors_odict)
```

Comment:
It is enough to save one file type. Please set the file extension at the same place with the other constants (regions, timeframe, carriers) so that it is easy to get an overview and being able to adapt that.

Reply:
I like that idea. I have done it in e993352.